src/simulation/simulator.h

```diff
@@ -64,36 +67,43 @@ public slots:
 	QChar getDeviceType (ItemBase*);
 	double getMaxPropValue(ItemBase*, QString);
 	QString getSymbol(ItemBase*, QString);
-	double getVectorValueOrDefault(const std::string & vecName, double defaultValue);
-	double calculateVoltage(ConnectorItem *, ConnectorItem *);
-	double getCurrent(ItemBase*, QString subpartName="");
-	double getTransistorCurrent(QString spicePartName, TransistorLeg leg);
-	double getPower(ItemBase*, QString subpartName="");
+    double getVectorValueOrDefault(unsigned long timeStep, const std::string & vecName,  double defaultValue);
+    double calculateVoltage(unsigned long, ConnectorItem *, ConnectorItem *);
+	std::vector<double> voltageVector(ConnectorItem *);
+    QString generateSvgPath(std::vector<double>, std::vector<double>, int, QString, double, double, double, double, double, double, double, double, QString, QString);
+    double getCurrent(unsigned long, ItemBase*, QString subpartName="");
+    double getTransistorCurrent(unsigned long timeStep, QString spicePartName, TransistorLeg leg);
+    double getPower(unsigned long, ItemBase*, QString subpartName="");
 
 	//Functions to update the parts
-	void updateCapacitor(ItemBase *);
-	void updateDiode(ItemBase *);
-	void updateLED(ItemBase *);
-	void updateMultimeter(ItemBase *);
-	void updateResistor(ItemBase *);
-	void updatePotentiometer(ItemBase *);
-	void updateDcMotor(ItemBase *);
-	void updateIRSensor(ItemBase *);
-	void updateBattery(ItemBase *);
+    void updateCapacitor(unsigned long, ItemBase *);
+    void updateDiode(unsigned long, ItemBase *);
+    void updateLED(unsigned long, ItemBase *);
+    void updateMultimeter(unsigned long, ItemBase *);
+    void updateOscilloscope(unsigned long, ItemBase *);
+    void updateResistor(unsigned long, ItemBase *);
+    void updatePotentiometer(unsigned long, ItemBase *);
+    void updateDcMotor(unsigned long, ItemBase *);
+    void updateIRSensor(unsigned long, ItemBase *);
+    void updateBattery(unsigned long, ItemBase *);
 
```

Comment:
Please set your editor to use single tabs, not fours spaces, for indentation.

Reply:
Done:
![image](https://github.com/user-attachments/assets/74a1545a-fc9a-453f-83ce-9c3ca9c9b688)